### PR TITLE
fix: tighten up byline data validation

### PR DESCRIPTION
**Pull request by aronshamash**
- Fixes issue with no bylines: https://nidigitalsolutions.jira.com/browse/REPLAT-6899
- Improve validation for byline data.
- adds `hasBylineData` as an export of `article-bylines` and now used on all article templates
- Fix some old fixture data that was out of date :( (need some TLC on fixtures....)

### CODE REVIEW

File: packages/article-in-depth/src/article-meta/has-byline-data.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export default bylines =>
+  bylines.length > 0 && bylines[0].byline && bylines[0].byline.length > 0;
```
Comment (apostolnikov) [inline packages/article-in-depth/src/article-meta/has-byline-data.js]: Do we have any reason not to add this inside the utils folder?

Comment (aronshamash) [reply packages/article-in-depth/src/article-meta/has-byline-data.js]: Well I was thinking utils is for more generic utility functions not something as specific as this. I'm more of the opinion to keep helper functions local to the package now, but I'm open to others opinion. @nathggns @nehasri89 @ValGeorgiev thoughts ?

**Comment by nathggns** [comment]
This will likely conflict with https://github.com/newsuk/times-components/pull/2101 – it might be better to wait for that to merge as yours will be easier to rebase than mine?

File: packages/article-in-depth/src/article-meta/has-byline-data.js (hunk 1 above)
Comment (nathggns) [reply packages/article-in-depth/src/article-meta/has-byline-data.js]: I'd put it in `article-skeleton` or `article-byline`.

File: packages/article-in-depth/src/article-meta/article-meta.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import Context from "@times-components/context";
 import DatePublication from "@times-components/date-publication";
 import { colours } from "@times-components/styleguide";
 
+import hasBylineData from "./has-byline-data";
```
Comment (nathggns) [inline packages/article-in-depth/src/article-meta/article-meta.js]: Can we please move the `hasBylineData` function from the new utils file in `@times-compoments/article-main-standard` into `article-byline`, upgrade that function, and share it across the different templates that way? As discussed on Slack .

File: packages/article-in-depth/src/article-meta/has-byline-data.js (hunk 1 above)
Comment (nathggns) [inline packages/article-in-depth/src/article-meta/has-byline-data.js]: Can we please move the hasBylineData function from the new utils file in @times-compoments/article-main-standard into article-byline, upgrade that function, and share it across the different templates that way? As discussed on Slack .

Comment (aronshamash) [reply packages/article-in-depth/src/article-meta/has-byline-data.js]: putting in `article-byline` and sharing

**Pull request merged**